### PR TITLE
[Refactor][Manifest] split elementwise.yaml into per-sub-family shards

### LINKED ISSUE #1093: [Refactor][Manifest] sub-split elementwise.yaml into unary/binary/multi-input
### Symptom / Motivation

Discovered during PR #1092 ([Refactor][Manifest] split monolithic yaml into per-family package).

After the family split, `tileops/manifest/elementwise.yaml` is 2030 lines / 62 ops — by far the largest family file (next is `reduction.yaml` at 922 lines / 19 ops, then `attention.yaml` at 568 lines / 12 ops). Elementwise alone is 54% of all manifest ops and ~44% of the manifest by line count, so the original "single big file" pain is partially recreated inside one family.

### Root Cause Analysis

Elementwise has a wide internal taxonomy — unary functional, binary, multi-input — each with distinct signature shapes (input arity, broadcasting rules, dtype constraints) but all sharing the same `family: elementwise` label. The PR #1092 split was intentionally scoped to "one file per family" to keep the diff (and review) reasonable; sub-splitting elementwise was explicitly deferred to a follow-up.

The loader (`tileops/manifest/__init__.py`) already merges any number of `*.yaml` files in the package and errors on duplicate keys, so adding more files is a pure data move with no code changes required.

### Related Files

- `tileops/manifest/elementwise.yaml`
- `tileops/manifest/__init__.py`

## Goal

Split `elementwise.yaml` by sub-category so no single manifest file exceeds ~1000 lines or ~30 ops.

## Plan

**Plan type: proposal**

1. Group the 62 elementwise ops by sub-category. Proposed cut (subject to refinement):
   - `elementwise_unary.yaml` — single-input, single-output (`AbsFwdOp`, `ExpFwdOp`, `LogFwdOp`, `SinFwdOp`, ...).
   - `elementwise_binary.yaml` — two-input, single-output (`AddFwdOp`, `MulFwdOp`, `BitwiseAndFwdOp`, ...).
   - `elementwise_multi_input.yaml` — three-or-more-input or multi-output (e.g., `WhereFwdOp`, fused triplets).
2. Move entries verbatim using `ruamel.yaml` round-trip; preserve `family: elementwise` on every entry (the sub-split is purely a file-level concern, not a semantic re-classification).
3. Verify `from tileops.manifest import load_manifest; assert len(load_manifest()) == 114` round-trips cleanly with no duplicate-key error.
4. Run `python scripts/validate_manifest.py` (must remain green) and `pytest tests/test_ops_manifest.py tests/test_validate_manifest.py tests/test_workloads_to_params.py`.

## Constraints

- Must not regress PR #1092: total op count stays at 114, no `family` field changes, public API (`load_workloads`) behavior identical.
- No op-code or kernel-code changes — pure manifest file partitioning.
- Same trust-model rules apply: this is a manifest-only PR; do not bundle op or kernel changes.

## Acceptance Criteria

- [ ] AC-1: `tileops/manifest/elementwise.yaml` is removed; ops live in 2-3 sub-files (`elementwise_unary.yaml` etc.) with no file exceeding ~1000 lines.
- [ ] AC-2: `load_manifest()` returns the same set of 114 op names with the same content as before the split (round-trip equality at the dict level).
- [ ] AC-3: `scripts/validate_manifest.py` passes with zero new errors.
- [ ] AC-4: `pytest tests/test_ops_manifest.py tests/test_validate_manifest.py tests/test_workloads_to_params.py` — same number passing as on `main` at branch point.

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #1093

## Summary

- Split `tileops/manifest/elementwise.yaml` (62 ops, ~2050 lines) into 4 sub-family shards by signature shape:
  - `elementwise_unary_math.yaml` — 24 ops (~746 lines): unary math (single-input single-output, math kernels)
  - `elementwise_unary_activation.yaml` — 13 ops (~485 lines): unary activations (single-input single-output, NN activations)
  - `elementwise_binary.yaml` — 23 ops (~739 lines): binary (two-input single-output)
  - `elementwise_multi_input.yaml` — 2 ops (~84 lines): 3+ input or multi-output
- Removed the monolithic `tileops/manifest/elementwise.yaml`; entries moved verbatim via `ruamel.yaml` round-trip, preserving `family: elementwise` on every entry.
- Updated `tileops/manifest/__init__.py` module docstring and `docs/manifest.md` / `CLAUDE.md` to describe the sharded layout.

No op-code or kernel-code changes — pure manifest file partitioning. Total op count stays at 114; `load_manifest()` / `load_workloads()` behavior is dict-equal to upstream/main.

## Test plan

- [x] AC-1: `tileops/manifest/elementwise.yaml` removed; ops live in 4 sub-files, every shard < 1000 lines and ≤ 30 ops (largest: 746 lines / 24 ops).
- [x] AC-2: `load_manifest()` returns the same 114 op names with dict-level content equal to upstream/main (verified via Python comparison: `name_sets_equal=True`, `content_equal=True`).
- [x] AC-3: `python scripts/validate_manifest.py` exits 0 — "All manifest checks passed" (58 warnings, unchanged from baseline).
- [x] AC-4: `pytest tests/test_ops_manifest.py tests/test_validate_manifest.py tests/test_workloads_to_params.py` → 198 passed, matching upstream/main baseline (198 passed) at branch point.
- [x] pre-commit passed on all changed files.

## Additional context

Follows the same trust-model rules as PR #1092 (which sharded the manifest into one-file-per-family): manifest-only PR, no op or kernel changes bundled. The elementwise family was the only family still over the ~1000-line / ~30-op soft cap after #1092 — this PR completes the per-sub-family layout.

### CODE REVIEW

File: tileops/manifest/elementwise_binary.yaml [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+  source:
+    kernel: tileops/kernels/elementwise.py
+    op: tileops/ops/elementwise.py
+    test: tests/ops/test_activation.py
+    bench: benchmarks/ops/bench_activation.py
+    bench_manifest_driven: false
+
+# ---------------------------------------------------------------------------
+# elementwise — multi-input parametric ops (where, clamp, masked_fill, nan_to_num)
+# ---------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:48]: Addressed in ae5ba0a.

Updated the comment to `# elementwise — two-input parametric ops (prelu, masked_fill)` to match the actual contents of the binary shard (PreluFwdOp + MaskedFillFwdOp; `where` / `clamp` / `nan_to_num` live in `elementwise_multi_input.yaml`).

Current HEAD: ae5ba0a

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+  source:
+    kernel: tileops/kernels/elementwise.py
+    op: tileops/ops/elementwise.py
+    test: tests/ops/test_special_elementwise.py
+    bench: benchmarks/ops/bench_independent_elementwise.py
+    bench_manifest_driven: false
+
+# ---------------------------------------------------------------------------
+# elementwise -- binary arithmetic ops (broadcast, output: same_as(input))
+# ---------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml]: Addressed in ae5ba0a.

Removed the trailing `# elementwise -- binary arithmetic ops` header — it was a leftover from the original monolithic `elementwise.yaml` and described content that is no longer in this file.

Current HEAD: ae5ba0a

File: tileops/manifest/elementwise_unary_math.yaml [hunk 3]
```diff
@@ -0,0 +1,746 @@
...
+    kernel: tileops/kernels/elementwise.py
+    op: tileops/ops/elementwise.py
+    test: tests/ops/test_special_elementwise.py
+    bench: benchmarks/ops/bench_unary_elementwise.py
+    bench_manifest_driven: false
+
+# ---------------------------------------------------------------------------
+# elementwise -- unary functional activations (torch.nn.functional.<op>)
+# All ops operate elementwise over arbitrary-rank tensors; output shape == input shape.
+# ---------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_math.yaml]: Addressed in ae5ba0a.

Removed the trailing `# elementwise -- unary functional activations` header — it was a leftover from the original monolithic `elementwise.yaml` and those ops now live in `elementwise_unary_activation.yaml`.

Current HEAD: ae5ba0a

File: tileops/manifest/__init__.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
 
-The manifest is split across one YAML file per op family in this package
-directory (e.g. ``elementwise.yaml``, ``reduction.yaml``). At load time all
-files are merged into a single ``ops`` dict; duplicate op names across files
-raise :class:`ValueError`.
+The manifest is split across one or more YAML files per op family in this
+package directory. Most families use a single file (e.g. ``reduction.yaml``),
+but a family MAY be sharded across multiple files when it grows large (e.g.
+the ``elementwise`` family). At load time all files are merged into a single
+``ops`` dict; duplicate op names across files raise :class:`ValueError`.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/__init__.py]: Addressed in 95f8989.

Applied your suggestion verbatim — dropped the RFC-style "MAY" wording and the ``reduction.yaml`` example to tighten the paragraph. Same information, fewer words.

Current HEAD: 95f8989

File: docs/design/manifest.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 
-The [`tileops/manifest/`](../tileops/manifest/) package is the **source of truth** for op interfaces, benchmark workloads, and roofline metadata.
+The [`tileops/manifest/`](../../tileops/manifest/) package is the **source of truth** for op interfaces, benchmark workloads, and roofline metadata.
 
 ## Layout
 
-The manifest is split across one YAML file per op family — `elementwise.yaml`, `reduction.yaml`, `attention.yaml`, `normalization.yaml`, `moe.yaml`, `scan.yaml`, `convolution.yaml`. Each file is a flat top-level mapping of op name → entry (no wrapper key). The `tileops.manifest` package merges all files at load time; duplicate op names across files are an error.
+The manifest is split across one or more YAML files per op family. A family is normally a single file; large families may be sharded across multiple files. Each file is a flat top-level mapping of op name → entry (no wrapper key). The `tileops.manifest` package merges all files at load time; duplicate op names across files are an error.
 
 - **Add or edit an op**: edit the family file matching the op's `family` field. Use `ruamel.yaml` for round-trip edits.
```
Comment (Ibuki-wind) [inline docs/design/manifest.md:9]: The new sharded-family layout makes `family` insufficient to identify the YAML file for an op: `elementwise` now has several files with the same family, so agents following this instruction can edit the wrong shard or fail to find the owner. → Change this guidance to locate the owning file by op key for existing ops (for example via `manifest_files()`/`load_manifest()` origin tracking or `rg '^<OpName>:' tileops/manifest/*.yaml`) and define how to choose a shard for new ops.

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
### Overall
Top risk is manifest edit guidance still assuming family uniquely identifies a YAML file; update that guidance for sharded families.

### Advisory
- This PR spans manifest, test, implementation, benchmark, docs, and validator surfaces across multiple trust-boundary stages; per migration policy, treated as advisory only.

**Review by Ibuki-wind — APPROVED** [review]
### Overall
Clean — no new issues in the unchanged SHA after re-evaluating the prior blocker under the confirmed design-doc scope.

### Prior blockers
All 1 prior blockers resolved — see thread replies by withdrawing the implementation-detail request from the design document.
